### PR TITLE
Create wallet - airdrop screens and logic

### PR DESCRIPTION
**Pull request by jellyfish-tom**
## What does this PR do?

Introduces create wallet process airdrop screens and logic around them

#### *Any scout cleaning/refactoring?*

some minor changes like empty lines or removed unused variables

## Todos:

- [X] Checked on iOS
- [ ] Checked on Android

### CODE REVIEW

File: src/screens/AirdropCreateWalletSubscription.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  route: RouteProp<RootStackParams, Route.AirdropCreateWalletSubscription>;
+  navigation: StackNavigationProp<RootStackParams, Route.AirdropCreateWalletSubscription>;
+  isLoading: boolean;
+  subscribeWallet: SubscribeWalletActionCreator;
+};
+
+const ShareComponent = () => (
+  <View style={styles.shareComponent}>
+    <Text style={styles.shareDescription}>
+      {i18n.formatString(i18n.airdrop.createWalletSuccess.shareIt, { rewardValue: 0.01 })}
```
Comment (KedziaPawel) [inline src/screens/AirdropCreateWalletSubscription.tsx:34]: move to const please

File: src/screens/AirdropCreateWalletSubscription.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    </Text>
+    <View style={styles.socialsContainer}>
+      <View style={styles.facebookButtonContainer}>
+        {/* TODO: fill share buttons content */}
+        <SocialShareFacebookButton shareOptions={{ message: 'Waiting for Content', title: 'Waiting for content' }} />
+      </View>
+      <SocialShareTwitterButton shareOptions={{ message: 'Waiting for Content', title: 'Waiting for content' }} />
+    </View>
+    <Text style={styles.maximumReward}>
+      {i18n.formatString(i18n.airdrop.createWalletSuccess.maxReward, { rewardValue: 0.02 })}
```
Comment (KedziaPawel) [inline src/screens/AirdropCreateWalletSubscription.tsx:44]: +1

File: src/screens/CreateWalletScreen.tsx [hunk 3]
```diff
@@ -112,7 +120,9 @@ export class CreateWalletScreen extends React.PureComponent<Props, State> {
           email,
           flowType: ConfirmAddressFlowType.SUBSCRIBE,
           walletsToSubscribe: [wallet],
-          onSuccess: this.navigateToSuccesfullNotificationSubscriptionMessage,
+          onSuccess: isAfterAirdrop()
+            ? this.navigateToSuccesfullNotificationSubscriptionMessage
+            : () => this.navigateToAirdropWalletSubscription(wallet, true),
```
Comment (KedziaPawel) [inline src/screens/CreateWalletScreen.tsx:125]: seems that missing logic, redirecting to airdrop if the user clicks `no` on notifications

Comment (jellyfish-tom) [reply src/screens/CreateWalletScreen.tsx:125]: not sure I understand but if user clicks NO on email notifications, it *should* still redirect to airdrop, only success message will be different at the end of wallet creation. 
What logic is missing?

File: src/components/ProgressButton.tsx [hunk 4]
```diff
@@ -66,7 +66,7 @@ export const ProgressButton: FC<Props> = ({
     return () => {
       clearInterval(intervalID);
     };
-  }, []);
+  }, [intervalID]);
```
[the remark this answers is not in the document]
Comment (jellyfish-tom) [reply src/components/ProgressButton.tsx:69]: ah, now I see it. Cant move styles outside as this component takes width, height and radius for button as parameters. 
Those styles are usage specific, so I believe it should stay like that, for the component to be generic.

File: src/screens/AirdropCreateWalletSubscription.tsx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+          buttonProps: {
+            title: i18n.formatString(i18n.airdrop.createWalletSuccess.successCompletedButton, {
+              routeName: parentRouteName === Route.AirdropDashboard ? i18n.airdrop.title : i18n.wallets.dashboard.title,
+            }),
+            onPress: () =>
+              parentRouteName === Route.AirdropDashboard
+                ? navigation.navigate(Route.AirdropDashboard)
+                : navigation.navigate(Route.MainTabStackNavigator, { screen: Route.Dashboard }),
+          },
+          shareComponent: <ShareComponent />,
```
[the remark this answers is not in the document]
Comment (jellyfish-tom) [reply src/screens/AirdropCreateWalletSubscription.tsx]: done

File: src/screens/AirdropCreateWalletSubscription.tsx (hunk 1 above)
Comment (jellyfish-tom) [reply src/screens/AirdropCreateWalletSubscription.tsx]: done

File: src/screens/AirdropCreateWalletSubscription.tsx (hunk 2 above)
Comment (jellyfish-tom) [reply src/screens/AirdropCreateWalletSubscription.tsx]: done

File: src/components/ProgressButton.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (KedziaPawel) [reply src/components/ProgressButton.tsx:69]: the styles that are not computed should be outside the component, the rest you can make inline styles

File: src/screens/CreateWalletScreen.tsx (hunk 3 above)
Comment (KedziaPawel) [reply src/screens/CreateWalletScreen.tsx:125]: yes exactly, and I don't see this navigating to airdrop if the user clicks "no" on notifications

File: src/components/ProgressButton.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jellyfish-tom) [reply src/components/ProgressButton.tsx:39]: done

File: src/screens/CreateWalletScreen.tsx (hunk 3 above)
Comment (jellyfish-tom) [reply src/screens/CreateWalletScreen.tsx:125]: done

File: src/screens/CreateWalletScreen.tsx [hunk 6]
```diff
@@ -112,22 +120,31 @@ export class CreateWalletScreen extends React.PureComponent<Props, State> {
...
         }),
-      onBack: () => this.props.navigation.navigate(Route.MainTabStackNavigator, { screen: Route.Dashboard }),
+      onBack: () =>
+        isAfterAirdrop()
+          ? this.props.navigation.navigate(Route.MainTabStackNavigator, { screen: Route.Dashboard })
+          : this.navigateToAirdropWalletSubscription(wallet, true),
```
Comment (KedziaPawel) [inline src/screens/CreateWalletScreen.tsx:130]: this.navigateToAirdropWalletSubscription(wallet)

Comment (jellyfish-tom) [reply src/screens/CreateWalletScreen.tsx:130]: a shit, true, thanks

Comment (jellyfish-tom) [reply src/screens/CreateWalletScreen.tsx]: done

**Review by KedziaPawel — APPROVED** [review]

**Pull request merged**